store/store.go

```diff
@@ -499,6 +506,67 @@ func (s *Store[H]) get(ctx context.Context, hash header.Hash) ([]byte, error) {
 	return data, nil
 }
 
+// notifyAndAdvance will notify waiters in heightSub and advance contiguousHead
+// based on already written headers.
+func (s *Store[H]) notifyAndAdvance(ctx context.Context, headers ...H) {
+	// always inform heightSub about new headers seen
+	for _, h := range headers {
+		s.heightSub.Notify(h.Height())
+	}
```

Comment:
Propose to make Notify variadic. This allows locking heightSub map once instead of relocking on every iteration. This is also how it was before with Pub.

Reply:
The only limitation here is that we're iterating over slice of headers, not heights. Done.

Reply:
True, so we either allocate a new slice or relock. Isnt as smooth as I expected

Reply:
IIRC this slice can be allocated on a stack rather then heap. But yeah, not as simple as we would like to see it.